Packages/Tracking Preview/PhysicsHands/Runtime/UI/Scripts/PhysicsButton.cs

```diff
@@ -0,0 +1,506 @@
+                enabled = false;
+            }
+
+            SetupButton();
+
+            _provider = FindObjectOfType<PhysicsProvider>(true);
+
+            OnPress += _OnPress.Invoke;
+            OnUnpress += _OnUnpress.Invoke;
+            OnHover += _OnHover.Invoke;
+            OnUnhover += _OnUnhover.Invoke;
+            OnContact += _OnContact.Invoke;
+            OnUncontact += _OnUncontact.Invoke;
+            OnPressedAmountChanged += _OnPressedAmountChanged.Invoke;
+            OnToggle += _OnToggle.Invoke;
+        }
+
+        protected void FixedUpdate()
+        {
+            if (_provider != null)
+            {
+                ProcessPhysicsEvents();
+            }
+        }
+
+        private void ProcessPhysicsEvents()
+        {
+            if (_provider.IsObjectHovered(_buttonElement.Rigid))
+            {
+                if (!_hovered)
+                {
+                    _hovered = true;
+                    OnHover?.Invoke();
+                }
+
+                if (_provider.GetObjectState(_buttonElement.Rigid, out var state))
+                {
+                    switch (state)
+                    {
+                        case PhysicsGraspHelper.State.Hover:
+                            _contactReleaseTime = Mathf.Clamp(--_contactReleaseTime, 0, 100);
+                            if (_contact && _contactReleaseTime == 0)
+                            {
+                                _contact = false;
+                                OnUncontact?.Invoke();
+                            }
+                            break;
+                        case PhysicsGraspHelper.State.Contact:
+                        case PhysicsGraspHelper.State.Grasp:
+                            if (!_contact)
+                            {
+                                _contact = true;
+                                OnContact?.Invoke();
+                                // Small wait to reduce erroneous uncontact events
+                                _contactReleaseTime = 10;
+                            }
+                            // Ensure that we only allow hands to press the button
+                            if (_handsOnly)
+                            {
+                                PressValidation();
+                            }
+                            break;
+                    }
+                }
+            }
+            else
+            {
+                if (_hovered)
+                {
+                    _hovered = false;
+                    OnUnhover?.Invoke();
+                }
+                if (_contact)
+                {
+                    _contact = false;
+                    OnUncontact?.Invoke();
+                }
+            }
+
+            // If we want all physics objects to press it then go wild
+            if (!_handsOnly)
+            {
+                PressValidation();
```

Comment:
If we are in hands only mode. then can anything set _pressedThisFrame?

Reply:
Yes, the hands will fire this event as per line 294. As press can only happen if the hands are in contact with the button we perform the check then.

Reply:
Thanks, I missed that.